### PR TITLE
[RSDK-4052] fix-parallel-projection-xy

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-4052)

1. Changes `transform.ParallelProjectionOntoXZWithRobotMarker` to `transform.ParallelProjectionOntoXYWithRobotMarker` 
2. This function was originally intended to be used by builtin SLAM during the map representation project and never was. 
3. Since that project we changed the dimensions the cartographer-module slam map is returned in from XZ to XY. This makes this function compatible with cartographer outputs.
4. Flips map on Y axis. I'm not sure why this was needed to make the jpeg match the three.js UI.
5. Applies the same color scheme in the three.js slam 2d renderer: https://github.com/viamrobotics/rdk/blob/main/web/frontend/src/components/slam/2d-renderer.svelte#L60
6. In the future we may want to consider moving `transform.ParallelProjectionOntoXYWithRobotMarker` out of RDK  as this function is very specific to cartographer-module & the way the RDK remote-control frontend currently happens to render pointclouds in 2d: https://viam.atlassian.net/browse/RSDK-4244

Output jpegs derived with:
```
go run pprojection/main.go ./get_point_cloud_map.pcd live.jpeg
```

[main.go.txt](https://github.com/viamrobotics/rdk/files/12176338/main.go.txt)

Example input (small):
[get_point_cloud_map.pcd.txt](https://github.com/viamrobotics/rdk/files/12174861/get_point_cloud_map.pcd.txt)
[position.json.txt](https://github.com/viamrobotics/rdk/files/12174867/position.json.txt)

 
Example output (small):
live5

Example output in three.js renderer (small):
Screenshot 2023-07-26 at 12 10 32 PM

Example input (large):
[pointcloud_23.pcd.txt](https://github.com/viamrobotics/rdk/files/12173529/pointcloud_23.pcd.txt)
[position_23.json.txt](https://github.com/viamrobotics/rdk/files/12174872/position_23.json.txt)
 
Example output (large):
artifact

Example output in three.js renderer (large):
Screenshot 2023-07-26 at 10 23 48 AM

### CODE REVIEW

File: pprojection/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package main
```
Comment (nicksanford) [inline pprojection/main.go:1]: delete before merge, this is just a convenient test harness to be able to iterate quickly on ensuring the resulting jpeg looks the way we want

File: rimage/transform/parallel_projection.go [hunk 2]
```diff
@@ -116,55 +116,56 @@ const (
...
 	}
 
 	// Calculate the scale factors
-	scaleFactor := calculateScaleFactor(maxX-minX, maxZ-minZ)
+	scaleFactor := calculateScaleFactor(maxX-minX, maxY-minY)
 
 	// Add points in the pointcloud to a new image
-	var pointColor rimage.Color
 	im := rimage.NewImage(imageWidth, imageHeight)
+	for i := 0; i < im.Width(); i++ {
```
Comment (nicksanford) [inline rimage/transform/parallel_projection.go:155]: sets the background to white

File: rimage/transform/parallel_projection.go [hunk 3]
```diff
@@ -116,55 +116,56 @@ const (
...
-		// the default color of white is used.
+		// the default color of black is used.
 		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight {
-			pointColor, err = getColorFromProbabilityValue(data)
-			if err != nil {
-				return false
-			}
-			im.Circle(image.Point{X: x, Y: y}, pointRadius, pointColor)
+			pointColor := getColorFromProbabilityValue(data)
+			im.Circle(image.Point{X: x, Y: flipY(y, imageHeight)}, pointRadius, pointColor)
```
Comment (nicksanford) [inline rimage/transform/parallel_projection.go:168]: flips y

File: rimage/transform/parallel_projection.go [hunk 4]
```diff
@@ -193,50 +194,26 @@ func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
...
-// a shade of blue if it is below the miss threshold. These shades will be more distinct the further from the threshold they are.
-func getColorFromProbabilityValue(d pointcloud.Data) (rimage.Color, error) {
-	var r, g, b uint8
-
-	if d == nil {
-		return rimage.NewColor(0, 0, 0), errors.New("data received was null")
+// getColorFromProbabilityValue returns an RGB color value based on the probability value
+// which is assumed to be in the blue color channel.
+// If no data or color is present, 100% probability is assumed.
+func getColorFromProbabilityValue(d pointcloud.Data) rimage.Color {
```
Comment (nicksanford) [inline rimage/transform/parallel_projection.go:204]: new color scheme based on probability

**Comment by EshaMaharishi** [comment]
Images look great! I'm taking myself off the PR so that John and I can split up some PR's. (I'm taking Kim's billing one.)

**Review by JohnN193 — APPROVED** [review]
LGTM, also fine with moving this into the cloud slam sessions repo since thats the only meaningful use case atm or the cartographer repo.

**Pull request merged**